src/output/scanline_output_file.rs

```diff
@@ -134,10 +159,73 @@ impl<'a> ScanlineOutputFile<'a> {
             let msg = unsafe { CStr::from_ptr(error_out) };
             Err(Error::Generic(msg.to_string_lossy().into_owned()))
         } else {
+            self.scanlines_written = self.header().data_dimensions().1;
```

Comment:
I'm pretty sure yes...?  If you write more pixel data than the image expects, I'm pretty sure that makes it an invalid file.  It certainly isn't useful, in any case.  Unless you can think of a use-case?

I mean, if we wanted to enforce this statically we could even have this method be self-consuming, and that would be semantically correct.  But I don't think that's necessary.